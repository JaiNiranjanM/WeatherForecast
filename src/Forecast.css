.next-date-details-cont {
  width: 98%;
  margin: auto;
  box-shadow: 0 3px 6px -6px black;
}

.details-temp {
  display: flex;
}

.details-temp-mid {
  margin: 0 3px;
}
