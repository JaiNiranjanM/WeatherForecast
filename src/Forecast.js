import React from "react";

import "./Forecast.css";

function Forecast(props) {
  const getFormatedData = () => {
    const days = [
      "Sunday",
      "Monday",
      "Tuesday",
      "Wednesday",
      "Thursday",
      "Friday",
      "Saturday",
    ];
    const date = new Date(props.availableDates);
    return `${days[date.getDay()]} ${date.getDate()}`;
  };

  const getWeatherForecastData = () => {
    const forecastArrayData = [];
    props.forecastData &&
      props.forecastData.list.filter((weather) => {
        if (
          weather &&
          weather.dt_txt &&
          weather.dt_txt.indexOf(props.availableDates) >= 0
        ) {
          forecastArrayData.push(weather);
        }
        return weather;
      });

    return forecastArrayData;
  };

  const selectedForecastArray = getWeatherForecastData();

  const getTime = (forecast) => {
    let finalTime = "";
    const time = forecast && forecast.dt_txt ? forecast.dt_txt : "";
    if (time) {
      const date = new Date(time);
      finalTime = date.getHours();
    }
    return finalTime === 0
      ? `12 am`
      : finalTime < 12
      ? `${finalTime} am`
      : finalTime > 12
      ? `${finalTime - 12} pm`
      : `${finalTime} pm`;
  };

  const getMinMaxTemp = (forecast) => {
    const minMaxTemperature = {
      minTemp:
        forecast && forecast.main && forecast.main.temp_min
          ? forecast.main.temp_min
          : "",
      maxTemp:
        forecast && forecast.main && forecast.main.temp_max
          ? forecast.main.temp_max
          : "",
    };
    return minMaxTemperature;
  };

  const getIconStatus = (forecast) => {
    return (
      forecast &&
      forecast.weather &&
      forecast.weather[0] &&
      forecast.weather[0].icon
    );
  };

  return (
    <div className="next-date-details-cont">
      <div className="date-details">{getFormatedData()}</div>
      <div className="weather-daily-details">
        {selectedForecastArray.map((forecast) => {
          return (
            <div className="details-data">
              <div className="details-time">{getTime(forecast)}</div>
              <div className="details-day">
                <img
                  src={`http://openweathermap.org/img/wn/${getIconStatus(
                    forecast
                  )}@2x.png`}
                  alt=""
                />
              </div>
              <div className="details-temp">
                <div>{getMinMaxTemp(forecast).minTemp.toFixed(1)}°C</div>
                <div className="details-temp-mid"> | </div>
                <div>{getMinMaxTemp(forecast).maxTemp.toFixed(1)}°C</div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
}

export default Forecast;
