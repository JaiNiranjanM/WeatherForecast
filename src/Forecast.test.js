import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import Forecast, { getIconStatus } from "./Forecast";

configure({ adapter: new Adapter() });

describe("Forecast Component", () => {
  const mockedIconStatus = jest.mock("./Forecast", () => ({
    getIconStatus: () => jest.fn(),
  }));

  it("should render Forecast component", () => {
    const component = shallow(<Forecast />);
    expect(component.getElements()).toMatchSnapshot();
  });

  test("Input should call handleChange on change event", () => {
    console.log("getIconStatus", getIconStatus);
    expect(jest.fn()).toBeCalledTimes(0);
  });
});
