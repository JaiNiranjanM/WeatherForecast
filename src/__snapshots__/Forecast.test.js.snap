// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Component should render Forecast component 1`] = `
Array [
  <div
    className="next-date-details-cont"
  >
    <div
      className="date-details"
    >
      undefined NaN
    </div>
    <div
      className="weather-daily-details"
    />
  </div>,
]
`;
